oak/server/storage/storage_node.cc

```diff
@@ -18,42 +17,102 @@
 #include "oak/server/storage/storage_node.h"
 
 #include "absl/memory/memory.h"
+#include "absl/strings/escaping.h"
+#include "asylo/util/cleansing_types.h"
 #include "asylo/util/logging.h"
 #include "grpcpp/create_channel.h"
 #include "oak/common/app_config.h"
 #include "oak/proto/storage_channel.pb.h"
 
 namespace oak {
 
+constexpr size_t kMaxMessageSize = 1 << 16;
+
 static std::string GetStorageId(const std::string& storage_name) {
   // TODO: Generate name-based UUID.
   return storage_name;
 }
 
+static asylo::CleansingVector<uint8_t> GetStorageEncryptionKey(const std::string& storage_name) {
+  // TODO: Request encryption key from escrow service.
+  std::string encryption_key =
+      absl::HexStringToBytes("c0dedeadc0dedeadc0dedeadc0dedeadc0dedeadc0dedeadc0dedeadc0dedead");
+  return asylo::CleansingVector<uint8_t>(encryption_key.begin(), encryption_key.end());
+}
+
 StorageNode::StorageNode(const std::string& name, const std::string& storage_address)
-    : NodeThread(name) {
+    : NodeThread(name),
+      fixed_nonce_generator_(new oak::FixedNonceGenerator()),
```

Comment:
I'm a bit nervous about the various dangling references that may hang around between storage node invocations &ndash; the `datum_name_cryptor_` references a `FixedNonceGenerator` which has a `string_view` to a `string` that was deleted some time ago.  Are the various objects lightweight enough that they could be generated on the fly for each storage operation (and thrown away after, reducing the chance of misuse)?

Reply:
Removed the string_view.

Reply:
How expensive are the `asylo::AesGcmSivCryptor` instances to construct?   I still idly wonder if they could be created/destroyed on the fly for storage operations, removing the need to hold them in `StorageNode` class (with `EncryptDatum`/`DecryptDatum` becoming `static` or anonymous `namespace` functions in (just) `storage_node.cc`).

Reply:
I'm not sure.  If we ever want to use sequential instead of random nonces it would make sense for them to remain here, but we can revisit that.